scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py

```diff
@@ -0,0 +1,211 @@
+
+        inputs["preRe"] = self.get_pre_inputs()
+
+        if "PhaseQuad" in self.view.getWS():
+            inputs['phaseTable'] = self.get_phase_table_inputs()
+
+        # do apodization and padding to complex data
+        if self.view.isComplex() and "PhaseQuad" not in self.view.getWS():
+            inputs['preIm'] = self.get_imaginary_inputs()
+
+        # do FFT to transformed data
+        inputs['FFT'] = self.get_fft_inputs()
+
+        return inputs
+
+    def get_input_run(self):
+        return self.getRun(
+            self.view.getInputWS().split(";", 1)[0].split("_", 1)[0])
+
+    def get_pre_inputs(self):
+        preInputs = self.view.initAdvanced()
+
+        if "PhaseQuad" in self.view.getWS():
+            preInputs['InputWorkspace'] = "__phaseQuad__"
+            preInputs['OutputWorkspace'] = "__ReTmp__"
+        else:
+            preInputs['OutputWorkspace'] = "__ReTmp__"
+            preInputs["InputWorkspace"] = self.clean(
+                self.view.getInputWS())
+
+        return preInputs
+
+    def get_phase_table_inputs(self):
+        phaseTable = {}
+        phaseTable["newTable"] = self.view.isNewPhaseTable()
+        phaseTable["FirstGoodData"] = self.view.getFirstGoodData()
+        phaseTable["LastGoodData"] = self.view.getLastGoodData()
+        phaseTable["Instrument"] = self.load.data_context.instrument
+        phaseTable["InputWorkspace"] = self.clean(
+            self.view.getInputWS())
+        phaseTable['MaskedDetectors'] = self.load.get_detectors_excluded_from_default_grouping_tables()
+
+        return phaseTable
+
+    def get_imaginary_inputs(self):
+        ImPreInputs = self.view.initAdvanced()
+        ImPreInputs['OutputWorkspace'] = "__ImTmp__"
+        ImPreInputs["InputWorkspace"] = self.clean(
+            self.view.getInputImWS())
+
+        return ImPreInputs
+
+    def get_fft_inputs(self):
+        FFTInputs = self.get_FFT_input()
+        if "PhaseQuad" in self.view.getWS():
+            self.view.getFFTRePhase(FFTInputs)
+            if self.view.isComplex():
+                self.view.getFFTImPhase(FFTInputs)
+
+            FFTInputs["OutputWorkspace"] = self.getRun(
+                self.view.getInputWS()) + ";PhaseQuad;FFT"
+        else:
+            FFTInputs["OutputWorkspace"] = self.getRun(
+                self.view.getInputWS()) + ";FFT"
+
+        return FFTInputs
+
+    # kills the thread at end of execution
+    def handleFinished(self):
+        self.activate()
+        self.thread.deleteLater()
+        self.thread = None
+
+    def get_FFT_input(self):
+        FFTInputs = self.view.initFFTInput(
+            self.clean(self.view.getInputWS()))
+
+        if self.view.isAutoShift():
+            FFTInputs["AutoShift"] = True
+        else:
+            self.view.addFFTShift(FFTInputs)
+        if self.view.isComplex():
+            self.view.addFFTComplex(FFTInputs)
+        return FFTInputs
+
+    def clean(self, name):
```

Comment:
can we change the name of this as it is a bit misleading? Maybe change it to extract_run?

Reply:
I agree the name is misleading. This is some legacy code which is removed in #25573 however so can we leave it as is for now.